### PR TITLE
[Refactoring] Docker Tika refactoring

### PR DESCRIPTION
**Pull request by Arsnael**
After realizing some issues with the work on Tika in an other refactoring PR (see reason here https://github.com/linagora/james-project/pull/2964#issuecomment-563658646), I preferred to extract the work done related to Tika extension creation and the singleton instance from #2964 and #2969 to enhance it here and make it cleaner.

I also took the freedom to migrate the test classes in mailbox-tika module to JUnit 5 (was small).

### CODE REVIEW

**Comment by Arsnael** [comment]
```
[7c4b42fd1a4528715c125c7301d0781112e70987] [ERROR] Failed to execute goal pl.project13.maven:git-commit-id-plugin:3.0.1:revision (get-the-git-infos) on project james-project: Execution get-the-git-infos of goal pl.project13.maven:git-commit-id-plugin:3.0.1:revision failed: Plugin pl.project13.maven:git-commit-id-plugin:3.0.1 or one of its dependencies could not be resolved: Could not transfer artifact com.google.guava:guava:jar:28.0-jre from/to central (https://repo.maven.apache.org/maven2): GET request of: com/google/guava/guava/28.0-jre/guava-28.0-jre.jar from central failed: Connection reset -> [Help 1]
```
test this please

File: mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaExtension.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    private final TikaContainer tika = TikaContainerSingleton.INSTANCE;
+
+    @Override
+    public void beforeAll(ExtensionContext extensionContext) {
+        tika.start();
+    }
+
+    @Override
+    public void afterAll(ExtensionContext extensionContext) {
+        tika.stop();
```
Comment (trantienduchn) [inline mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaExtension.java]: do you really want to stop the singleton ?

File: mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaExtension.java [hunk 2]
```diff
@@ -27,11 +27,7 @@
 import org.junit.jupiter.api.extension.ParameterResolver;
 
 public class TikaExtension implements AfterAllCallback, BeforeAllCallback, ParameterResolver {
-    private final TikaContainer tika;
-
-    public TikaExtension() {
-        this.tika = new TikaContainer();
-    }
+    private final TikaContainer tika = TikaContainerSingleton.INSTANCE;
```
Comment (chibenwa) [inline mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaExtension.java]: Stop calling start / stop in beforeAll / afterAll

File: mailbox/scanning-search/src/test/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndexTest.java [hunk 3]
```diff
@@ -54,192 +54,192 @@ protected void initializeMailboxManager() {
...
     }
 
-    @Ignore
+    @Disabled
     @Override
     public void searchWithTextShouldReturnMailsWhenHtmlBodyMatchesWithStemming() throws Exception {
     }
 
-    @Ignore
+    @Disabled
```
Comment (chibenwa) [inline mailbox/scanning-search/src/test/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndexTest.java]: While we are at it, is there a ticket for these failed tests? 

Can we put it in the description?

Comment (Arsnael) [reply mailbox/scanning-search/src/test/java/org/apache/james/mailbox/store/search/SimpleMessageSearchIndexTest.java]: Good point, will try to look for it...

File: mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaExtension.java (hunk 2 above)
Comment (Arsnael) [reply mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaExtension.java]: I totally get that, but I got confused when looking in other extensions (like Cassandra or Elasticsearch) where we totally do that... So we agree this behavior is wrong, and we might need to change it at some point for other docker extensions using singletons as well?

File: mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndexTest.java [hunk 4]
```diff
@@ -56,92 +55,92 @@ protected void initializeMailboxManager() {
         messageSearchIndex = resources.getSearchIndex();
     }
 
-    @Disabled
+    @Disabled("JAMES-1799")
```
Comment (chibenwa) [inline mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndexTest.java]: Can we add a little text explanation?

File: mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndexTest.java [hunk 5]
```diff
@@ -55,92 +55,92 @@ protected void initializeMailboxManager() {
         messageSearchIndex = resources.getSearchIndex();
     }
 
-    @Disabled("JAMES-1799")
+    @Disabled("JAMES-1799: ignoring failing test after port of ES tests to Lucene")
```
Comment (chibenwa) [inline mailbox/lucene/src/test/java/org/apache/james/mailbox/lucene/search/LuceneMessageSearchIndexTest.java]: `JAMES-1799: ignoring failing test after generalizing ElasticSearch test suite to other mailbox search backends` ?

**Comment by Arsnael** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by Arsnael** [comment]
```
[54c19b9adc4f3e717146b0c769c878c3aec76b8d] [ERROR]   EventDeadLettersIntegrationTest.failedEventShouldNotBeInDeadLettersAfterSuccessfulRedelivery:334->retrieveFirstFailedInsertionId:197 » IndexOutOfBounds
```
Wishing Duc's fix was merged already... :)

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**